content/ja/blog/_posts/2024-04-05-diy-create-your-own-cloud-with-kubernetes-part-1/index.md

```diff
@@ -0,0 +1,219 @@
+
+私たちのことですか？
+Ænixでは、ZFS、DRBD、OpenvSwitchなどのかなり特殊なカーネルモジュールを使用しているので、必要なモジュールをすべて事前に含んだシステムイメージを形成する方法を選びました。
+この場合、Talos Linuxが私たちにとって最も便利であることがわかりました。
+たとえば、次のような設定で、必要なカーネルモジュールをすべて含むシステムイメージを構築するのに十分です:
+
+```yaml
+arch: amd64
+platform: metal
+secureboot: false
+version: v1.6.4
+input:
+  kernel:
+    path: /usr/install/amd64/vmlinuz
+  initramfs:
+    path: /usr/install/amd64/initramfs.xz
+  baseInstaller:
+    imageRef: ghcr.io/siderolabs/installer:v1.6.4
+  systemExtensions:
+    - imageRef: ghcr.io/siderolabs/amd-ucode:20240115
+    - imageRef: ghcr.io/siderolabs/amdgpu-firmware:20240115
+    - imageRef: ghcr.io/siderolabs/bnx2-bnx2x:20240115
+    - imageRef: ghcr.io/siderolabs/i915-ucode:20240115
+    - imageRef: ghcr.io/siderolabs/intel-ice-firmware:20240115
+    - imageRef: ghcr.io/siderolabs/intel-ucode:20231114
+    - imageRef: ghcr.io/siderolabs/qlogic-firmware:20240115
+    - imageRef: ghcr.io/siderolabs/drbd:9.2.6-v1.6.4
+    - imageRef: ghcr.io/siderolabs/zfs:2.1.14-v1.6.4
+output:
+  kind: installer
+  outFormat: raw
+```
+
+`docker`コマンドラインツールを使用して、OSイメージをビルドします:
+
+```
+cat config.yaml | docker run --rm -i -v /dev:/dev --privileged "ghcr.io/siderolabs/imager:v1.6.4" -
+```
+
+その結果、必要なものがすべて含まれたDockerコンテナイメージが得られます。
+このイメージを使用して、サーバーにTalos Linuxをインストールできます。
+同じことができます。このイメージには、必要なすべてのファームウェアとカーネルモジュールが含まれます。
+
+しかし、新しく形成されたイメージをノードにどのように配信するかという問題が発生します。
+
+しばらくの間、PXEブートのアイデアについて考えていました。
+たとえば、2年前に[記事](/blog/2021/12/22/kubernetes-in-kubernetes-and-pxe-bootable-server-farm/)を書いた**Kubefarm**プロジェクトは、完全にこのアプローチを使用して構築されました。
+しかし残念ながら、他のクラスターを保持する最初の親クラスターをデプロイするのに役立つわけではありません。
+そこで今回、PXEアプローチを使用して同じことを行うのに役立つソリューションを用意しました。
+
+基本的に必要なのは、コンテナ内で一時的な**DHCP**と**PXE**サーバーを[実行する](https://cozystack.io/docs/get-started/)ことだけです。
+そうすれば、ノードはあなたのイメージから起動し、Debianベースの簡単なスクリプトを使用して、ノードのブートストラップに役立てることができます。
+
+[![asciicast](asciicast.svg)](https://asciinema.org/a/627123)
+
+`talos-bootstrap`スクリプトの[ソースコード](https://github.com/aenix-io/talos-bootstrap/)はGitHubで入手できます。
+
+このスクリプトを使用すると、ベアメタル上に5分でKubernetesをデプロイし、それにアクセスするためのkubeconfigを取得できます。
+しかし、まだ多くの未解決の問題が残っています。
+
+## システムコンポーネントの配信
+
+この段階では、さまざまなワークロードを実行できるKubernetesクラスターがすでに手に入っています。
+しかし、まだ完全に機能しているわけではありません。
+つまり、ネットワークとストレージを設定する必要があるだけでなく、仮想マシンを実行するためのKubeVirtや、監視スタックやその他のシステム全体のコンポーネントなど、必要なクラスター拡張機能をインストールする必要があります。
+
+従来、これは**Helmチャート**をクラスターにインストールすることで解決されています。
+ローカルで`helm install`コマンドを実行することで実現できますが、アップデートを追跡したい場合や、複数のクラスターを持っていてそれらを均一に保ちたい場合、このアプローチは不便になります。
+実際には、これを宣言的に行う方法はたくさんあります。
+これを解決するには、最高のGitOpsプラクティスを使用することをお勧めします。
+つまり、ArgoCDやFluxCDのようなツールを指します。
+
+ArgoCDは、グラフィカルインターフェースと中央コントロールプレーンを備えているため、開発目的には便利ですが、一方でFluxCDは、Kubernetes ディストリビューションの作成により適しています。
+FluxCDを使用すると、どのチャートをどのパラメーターで起動すべきかを指定し、依存関係を記述できます。
+そうすれば、FluxCDがすべてを処理してくれます。
+
+新しく作成したクラスターにFluxCDを1回インストールし、適切に設定することをお勧めします。
+これにより、FluxCDは必要不可欠なコンポーネントをすべて自動的にデプロイできるようになり、クラスターを目的の状態にアップグレードできます。
+たとえば、私たちのプラットフォームをインストールすると、システムコンポーネントとともに次の事前設定されたHelmチャートが表示されます:
```

Comment:
以下の原文を直訳すると、内容が重複するため、2つの文章の一部を統合して翻訳しました。

```
It is suggested to perform a one-time installation of FluxCD in your newly created cluster and provide it with the configuration. This will install everything necessary, bringing the cluster to the expected state.

By carrying out a single installation of FluxCD in your newly minted cluster and configuring it accordingly, you enable it to automatically deploy all the essentials. This will allow your cluster to upgrade itself into the desired state. For example, after installing our platform you'll see the next pre-configured Helm charts with system components:
```

Reply:
これ確かに気になりますね。。upstreamに質問しても良いかもしれないですね

Reply:
> upstreamに質問しても良いかもしれないですね
これは、Issueを起票して質問したら宜しいでしょうか？？

Reply:
Slackの`#sig-docs`チャンネルで聞くくらいで良いかなとは思いました。

とはいえこのPRのスコープ外なので一旦進めますね